src/main/java/com/meilisearch/sdk/SearchRequest.java

```diff
@@ -326,174 +330,21 @@ private SearchRequest(
-        return this;
-    }
-
-    /**
-     * Method to set the cropMarker
-     *
-     * @param cropMarker Marker used to crop field values
-     * @return altered SearchRequest
-     */
-    public SearchRequest setCropMarker(String cropMarker) {
-        this.cropMarker = cropMarker;
-        return this;
-    }
-
-    /**
-     * Method to set the highlightPreTag
+     * <p>This method is an alias of setQ()
      *
-     * @param highlightPreTag Highlight tag use before every highlighted query terms
-     * @return altered SearchRequest
-     */
-    public SearchRequest setHighlightPreTag(String highlightPreTag) {
-        this.highlightPreTag = highlightPreTag;
-        return this;
-    }
-
-    /**
-     * Method to set the highlightPostTag
-     *
-     * @param highlightPostTag Highlight tag use after every highlighted query terms
-     * @return altered SearchRequest
-     */
-    public SearchRequest setHighlightPostTag(String highlightPostTag) {
-        this.highlightPostTag = highlightPostTag;
-        return this;
-    }
-
-    /**
-     * Method to set the attributesToHighlight
-     *
-     * @param attributesToHighlight Attributes whose values will contain highlighted matching terms
-     * @return altered SearchRequest
-     */
-    public SearchRequest setAttributesToHighlight(String[] attributesToHighlight) {
-        this.attributesToHighlight = attributesToHighlight;
-        return this;
-    }
-
-    /**
-     * Method to set the filter
-     *
-     * @param filter Filter queries by an attribute value
-     * @return altered SearchRequest
-     */
-    public SearchRequest setFilter(String[] filter) {
-        if (filter != null) {
-            this.filter = filter;
-        }
-        return this;
-    }
-
-    public SearchRequest setFilterArray(String[][] filterArray) {
-        if (filterArray != null) {
-            this.filterArray = filterArray;
-        }
-        return this;
-    }
-
-    /**
-     * Method to set the matches boolean
-     *
-     * @param matches Defines whether an object that contains information about the matches should
-     *     be returned or not
-     * @return altered SearchRequest
-     */
-    public SearchRequest setMatches(boolean matches) {
-        this.matches = matches;
-        return this;
-    }
-
-    /**
-     * Method to set the facetsDistribution
-     *
-     * @param facetsDistribution Facets for which to retrieve the matching count
-     * @return altered SearchRequest
-     */
-    public SearchRequest setFacetsDistribution(String[] facetsDistribution) {
-        this.facetsDistribution = facetsDistribution;
-        return this;
-    }
-
-    /**
-     * Method to set the sort
-     *
-     * @param sort Sort queries by an attribute value
-     * @return altered SearchRequest
+     * @param q Query String
      */
-    public SearchRequest setSort(String[] sort) {
-        this.sort = sort;
-        return this;
+    public void setQuery(String q) {
```

Comment:
@alallema, I saw you marked it as resolved, but the code should be: `public SearchRequest setQuery(String q) {` doesn't it?

Reply:
<img width="510" alt="Screenshot 2022-11-08 at 08 27 55" src="https://user-images.githubusercontent.com/16155041/200501009-222704e1-3484-46bc-a03c-0696a0a6185b.png">
Because for me it is ... It was modify on this [commit](https://github.com/meilisearch/meilisearch-java/pull/425/commits/6840d4ad7bddd659c9f2632e80da365d359500b2)

Reply:
Awesome, I can see it now!